AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt

```diff
@@ -1009,180 +1018,218 @@ open class CardBrowser : NavigationDrawerActivity(), SubtitleListener, DeckSelec
         // dismiss undo-snackbar if shown to avoid race condition
         // (when another operation will be performed on the model, it will undo the latest operation)
         if (mUndoSnackbar != null && mUndoSnackbar!!.isShown) mUndoSnackbar!!.dismiss()
-        val itemId = item.itemId
-        if (itemId == android.R.id.home) {
-            endMultiSelectMode()
-            return true
-        } else if (itemId == R.id.action_add_note_from_card_browser) {
-            addNoteFromCardBrowser()
-            return true
-        } else if (itemId == R.id.action_save_search) {
-            val searchTerms = mSearchView!!.query.toString()
-            showDialogFragment(
-                newInstance(
-                    null, mMySearchesDialogListener,
-                    searchTerms, CardBrowserMySearchesDialog.CARD_BROWSER_MY_SEARCHES_TYPE_SAVE
+        when (item.itemId) {
```

Comment:
```suggestion
       return when (item.itemId) {
```
and we can remove all other `return's`

Reply:
Got to be a better way - leaving a cleanup to think about it